src/assets/Standards/ERC20Standard.ts

```diff
@@ -0,0 +1,93 @@
+import { abiERC20 } from '@metamask/metamask-eth-abis';
+import { BN } from 'ethereumjs-util';
+import { ERC20 } from '../../constants';
+
+export class ERC20Standard {
+  private web3: any;
+
+  constructor(web3?: any) {
+    this.web3 = web3;
+  }
+
+  /**
+   * Get balance or count for current account on specific asset contract.
+   *
+   * @param address - Asset ERC20 contract address.
+   * @param selectedAddress - Current account public address.
+   * @returns Promise resolving to BN object containing balance for current account on specific asset contract.
+   */
+  async getBalanceOf(address: string, selectedAddress: string): Promise<BN> {
+    const contract = this.web3.eth.contract(abiERC20).at(address);
+    return new Promise<BN>((resolve, reject) => {
+      contract.balanceOf(selectedAddress, (error: Error, result: BN) => {
+        /* istanbul ignore if */
+        if (error) {
+          reject(error);
+          return;
+        }
+        resolve(result);
+      });
+    });
+  }
+
+  /**
+   * Query for the decimals for a given ERC20 asset.
+   *
+   * @param address - ERC20 asset contract string.
+   * @returns Promise resolving to the 'decimals'.
+   */
+  async getTokenDecimals(address: string): Promise<string> {
+    const contract = this.web3.eth.contract(abiERC20).at(address);
+    return new Promise<string>((resolve, reject) => {
+      contract.decimals((error: Error, result: string) => {
+        /* istanbul ignore if */
+        if (error) {
+          reject(error);
+          return;
+        }
+        resolve(result);
+      });
+    });
+  }
+
+  /**
+   * Query for symbol for a given ERC20 asset.
+   *
+   * @param address - ERC20 asset contract address.
+   * @returns Promise resolving to the 'symbol'.
+   */
+  async getTokenSymbol(address: string): Promise<string> {
+    const contract = this.web3.eth.contract(abiERC20).at(address);
+    return new Promise<string>((resolve, reject) => {
+      contract.symbol((error: Error, result: string) => {
+        /* istanbul ignore if */
+        if (error) {
+          reject(error);
+          return;
+        }
+        resolve(result);
+      });
+    });
+  }
+
+  /**
+   * Query if a contract implements an interface.
+   *
+   * @param address - Asset contract address.
+   * @param userAddress - The public address for the currently active user's account.
+   * @returns Promise resolving an object containing the standard, decimals, symbol and balance of the given contract/userAddress pair.
+   */
+  async getDetails(address: string, userAddress: string) {
```

Comment:
What do you think about providing a return type for this?

Reply:
done here: https://github.com/MetaMask/controllers/pull/667/commits/8effcafddb6ca32d9566050ba50db569ceb00f07